### PR TITLE
Disallow multiple clicks on editor button

### LINKED ISSUE #2214: Editor button shouldn't be clickable multiple times
just in case the transition to the editor can cause problems, pressing the button more than once (per activation of the button) shouldn't be allowed. So once pressed it should move to the unpressable state.

### PR DESCRIPTION
**Pull request by Kasterisk**
Closes #2214

### CODE REVIEW

File: src/microbe_stage/MicrobeHUD.cs [hunk 1]
```diff
@@ -465,6 +467,9 @@ public void EditorButtonPressed()
     {
         GD.Print("Move to editor pressed");
 
+        // To prevent being clicked twice
+        editorButton.MouseFilter = Control.MouseFilterEnum.Ignore;
```
Comment (hhyyrylainen) [inline src/microbe_stage/MicrobeHUD.cs:471]: Wouldn't it be better to go to "Disabled" state instead?

Comment (Kasterisk) [reply src/microbe_stage/MicrobeHUD.cs:471]: Actually yeah thath makes sense. Changed.

**Comment by hhyyrylainen** [comment]
Noticed one tiny bug: in freebuild the button gets immediately re-enabled and plays the sound again. So perhaps the freebuild button enable should check that the game is not currently transitioning to the editor?

**Comment by Kasterisk** [comment]
Fixed that one.

File: src/microbe_stage/MicrobeStage.cs [hunk 2]
```diff
@@ -543,7 +551,11 @@ private void OnPlayerReproductionStatusChanged(Microbe player, bool ready)
         if (ready)
         {
             TutorialState.SendEvent(TutorialEventType.MicrobePlayerReadyToEdit, EventArgs.Empty, this);
-            HUD.ShowReproductionDialog();
+
+            // This is to prevent the editor button being able to be clicked
+            // multiple times in freebuild mode
```
Comment (hhyyrylainen) [inline src/microbe_stage/MicrobeStage.cs:556]: Doesn't this comment fit in a single line within 120 characters?

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
Just one minor nitpick otherwise this is good.

**Review by hhyyrylainen — APPROVED** [review]

**Pull request merged**